File: Projects/DataProviderApp/app/src/main/java/com/example/dataproviderapp/dto/responses/GpsDataDto.kt
package com.example.dataproviderapp.dto.responses

import java.time.LocalDateTime

data class GpsDataDto(
    val recId: ULong,
    val recDatetime: LocalDateTime,
    val tripId: ULong,
    val latitudeDeg: Float,
    val longitudeDeg: Float,
    val accuracyM: Float?,
    val speedKmh: UInt?,
    val bearingDeg: UShort?
)

File: Projects/server/Contracts/Responses/GpsDataDto.cs
namespace server.Contracts.Responses
{
    public class GpsDataDto
    {
        public ulong RecId { get; set; }
        public DateTime RecDatetime { get; set; }
        public ulong TripId { get; set; }
        public float LatitudeDeg { get; set; }
        public float LongitudeDeg { get; set; }
        public float? AccuracyM { get; set; }
        public uint? SpeedKmh { get; set; }
        public float? BearingDeg { get; set; }
    }
}
